packages/webvh/src/anoncreds/services/WebVhAnonCredsRegistry.ts

```diff
@@ -479,32 +486,18 @@ export class WebVhAnonCredsRegistry implements AnonCredsRegistry {
         return false
       }
 
-      // For eddsa-jcs-2022, we need to:
-      // 1. Create a document with the content and proof metadata (without proofValue)
-      // 2. Canonicalize it using JCS (JSON Canonicalization Scheme)
-      // 3. Verify the signature using EdDSA
-
-      // Create the document to be verified (content + proof without proofValue)
-      const documentToVerify = {
-        ...(content as Record<string, unknown>),
-        proof: {
-          type: proof.type,
-          cryptosuite: proof.cryptosuite,
-          verificationMethod: proof.verificationMethod,
-          proofPurpose: proof.proofPurpose || 'assertionMethod',
-          created: proof.created || new Date().toISOString(),
-        },
-      }
-
-      // Canonicalize the document using JCS
-      const canonicalizedDocument = canonicalize(documentToVerify)
-      const documentBytes = TypedArrayEncoder.fromString(canonicalizedDocument)
+      // For eddsa-jcs-2022 hashing algorithm:
+      // https://www.w3.org/TR/vc-di-eddsa/#hashing-eddsa-jcs-2022
+      const proofConfig = (({ proofValue, ...o }) => o)(proof);
+      const proofConfigHash = HashData(canonicalize(proofConfig));
+      const transformedDocumentHash = HashData(canonicalize(unsecuredDocument));
+      const hashData = proofConfigHash+transformedDocumentHash;
```

Comment:
I'm not completely sure about the procedure you need to achieve with the hashes but I think this code may help you to reach it without any external dependency:

```suggestion
      const proofConfigHash = Hasher.hash(canonicalize(proofConfig), 'sha-256')
      const transformedDocumentHash = Hasher.hash(canonicalize(unsecuredDocument), 'sha-256')
      const hashData = Buffer.concat([proofConfigHash, transformedDocumentHash])
```

You just need to import Hasher from @credo-ts/core.

Reply:
thanks, I ended up importing the Sha256 hasher directly from the @credo-ts/core